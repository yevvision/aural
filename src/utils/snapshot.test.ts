import { describe, it, expect } from 'vitest';

// Funktionen fÃ¼r Snapshot-Tests
export function createUserProfile(name: string, age: number, email: string) {
  return {
    id: 'fixed-id-123',
    name,
    age,
    email,
    createdAt: '2025-01-01T00:00:00.000Z',
    preferences: {
      theme: 'dark',
      notifications: true,
      language: 'de'
    }
  };
}

export function formatTrackData(track: any) {
  return {
    id: track.id,
    title: track.title,
    artist: track.artist,
    duration: `${Math.floor(track.duration / 60)}:${(track.duration % 60).toString().padStart(2, '0')}`,
    genre: track.genre || 'Unknown',
    year: track.year || new Date().getFullYear()
  };
}

describe('Snapshot Tests', () => {
  it('creates consistent user profile', () => {
    const profile = createUserProfile('Max Mustermann', 25, 'max@example.com');
    expect(profile).toMatchSnapshot();
  });

  it('formats track data consistently', () => {
    const track = {
      id: 'track-123',
      title: 'Test Song',
      artist: 'Test Artist',
      duration: 180,
      genre: 'Electronic'
    };
    
    const formatted = formatTrackData(track);
    expect(formatted).toMatchSnapshot();
  });

  it('handles missing track data', () => {
    const track = {
      id: 'track-456',
      title: 'Minimal Track',
      artist: 'Unknown Artist',
      duration: 120
    };
    
    const formatted = formatTrackData(track);
    expect(formatted).toMatchSnapshot();
  });
});
