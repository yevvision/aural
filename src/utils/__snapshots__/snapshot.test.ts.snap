// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Tests > creates consistent user profile 1`] = `
{
  "age": 25,
  "createdAt": "2025-01-01T00:00:00.000Z",
  "email": "max@example.com",
  "id": "fixed-id-123",
  "name": "Max Mustermann",
  "preferences": {
    "language": "de",
    "notifications": true,
    "theme": "dark",
  },
}
`;

exports[`Snapshot Tests > formats track data consistently 1`] = `
{
  "artist": "Test Artist",
  "duration": "3:00",
  "genre": "Electronic",
  "id": "track-123",
  "title": "Test Song",
  "year": 2025,
}
`;

exports[`Snapshot Tests > handles missing track data 1`] = `
{
  "artist": "Unknown Artist",
  "duration": "2:00",
  "genre": "Unknown",
  "id": "track-456",
  "title": "Minimal Track",
  "year": 2025,
}
`;
